src/slackapp/flows/createDeliveryRequest.js

```diff
@@ -0,0 +1,379 @@
+  slackInteractions.viewSubmission(sendMessage.id, guard(sendMessage));
+};
+
+/**
+ * Presents a new modal with a selector for all open requests (that aren't on slack already)
+ */
+async function selectRequestForSending(payload) {
+  let [requests, err] = await findOpenRequests(); // eslint-disable-line
+  requests = sortBy(requests, r => r.get("Last Modified")).reverse();
+
+  let view = null;
+  if (err) {
+    view = errorView(`Error looking up pending requests: ${err}`);
+  } else if (requests.length === 0) {
+    view = errorView(
+      `Couldn't find any pending requests. Is your request in the Delivery Needed state?`
+    );
+  } else {
+    view = await makeRequestSelectionView(requests);
+  }
+  await slackapi.views.open({
+    trigger_id: payload.trigger_id,
+    view
+  });
+}
+
+/**
+ * Makes a suggestested message and allows the user to edit it and choose which channel to send to.
+ */
+async function draftRequest(payload) {
+  const code =
+    payload.view.state.values.select_request.selected_request.selected_option
+      .value;
+  const [request, err] = await findRequestByCode(code);
+  if (err) {
+    return errorResponse(`Request ${code} wasn't found. ${err}`);
+  }
+  const form = await makeRequestDraftView(payload, request);
+  return {
+    response_action: "update",
+    view: form
+  };
+}
+
+/**
+ * Renders the message as markdown and lets the user confirm or go back to editing
+ */
+async function draftConfirm(payload) {
+  const code = payload.view.private_metadata;
+  const { values } = payload.view.state;
+  const content = values.draft_message.draft_message.value;
+  const channel = values.selected_channel.selected_channel.selected_channel;
+  const [request, err] = await findRequestByCode(code);
+  if (err) {
+    return errorResponse(`Request ${code} wasn't found. ${err}`);
+  }
+  return {
+    response_action: "push",
+    view: makeConfirmationView(request, channel, content)
+  };
+}
+
+/**
+ * Posts the message to slack. The bot also sends a DM to the user with a link to the message
+ * and updates the Request with the metadata for the message.
+ */
+async function sendMessage(payload) {
+  const context = JSON.parse(payload.view.private_metadata);
+
+  // Send the message
+  const deliveryMessage = await slackapi.chat.postMessage({
+    channel: context.channelId,
+    text: context.content
+  });
+
+  // Update the metadata on the Request
+  await updateRequestByCode(context.code, {
+    Status: "Delivery Needed",
+    Meta: {
+      slack_ts: deliveryMessage.ts,
+      slack_channel: deliveryMessage.channel
+    }
+  });
+
+  // DM the user with a permalink
+  const permalink = await slackapi.chat.getPermalink({
+    channel: deliveryMessage.channel,
+    message_ts: deliveryMessage.ts
+  });
+  slackapi.chat.postMessage({
+    channel: payload.user.id,
+    text: `*${context.code}*: Submitted a new request for delivery to <#${deliveryMessage.channel}>\n${permalink.permalink}`,
```

Comment:
nit: submitted -> posted

Reply:
Removed permalink entirely anyway as you suggested.